### PR TITLE
Add `uv tool dir --bin` to show executable directory

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/uv/issues/5159.

### CODE REVIEW

File: crates/uv/src/commands/tool/dir.rs [hunk 1]
```diff
@@ -3,16 +3,23 @@ use owo_colors::OwoColorize;
...
-    let installed_tools =
-        InstalledTools::from_settings().context("Failed to initialize tools settings")?;
-    anstream::println!("{}", installed_tools.root().simplified_display().cyan());
+
+    if bin {
+        let executable_directory = find_executable_directory()?;
+        anstream::println!("{}", executable_directory.simplified_display().cyan());
+    } else {
+        let installed_tools =
+            InstalledTools::from_settings().context("Failed to initialize tools settings")?;
```
Comment (zanieb) [inline crates/uv/src/commands/tool/dir.rs:20]: Fyi I think this is infallible. The error case comes from `StateStore::from_path` which uses a `Result` but returns an `Err`?

File: crates/uv/src/commands/tool/dir.rs [hunk 2]
```diff
@@ -3,16 +3,23 @@ use owo_colors::OwoColorize;
...
     if preview.is_disabled() {
         warn_user_once!("`uv tool dir` is experimental and may change without warning.");
     }
-    let installed_tools =
-        InstalledTools::from_settings().context("Failed to initialize tools settings")?;
-    anstream::println!("{}", installed_tools.root().simplified_display().cyan());
+
+    if bin {
+        let executable_directory = find_executable_directory()?;
+        anstream::println!("{}", executable_directory.simplified_display().cyan());
```
Comment (zanieb) [inline crates/uv/src/commands/tool/dir.rs:17]: Is there a reason you opted for the full name here? I think it's standard to do

```
use anstream::{eprintln, println};
```

nearly everywhere else.

File: crates/uv-cli/src/lib.rs [hunk 3]
```diff
@@ -2168,6 +2168,17 @@ pub struct ToolInstallArgs {
...
 pub struct ToolListArgs;
 
+#[derive(Args)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct ToolDirArgs {
+    /// Show the directory into which `uv tool` will install executables.
+    ///
+    /// By default, `uv tool dir` shows the directory into which the tool Python environments
+    /// themselves are installed, rather than the directory containing the linked executables.
+    #[arg(long, alias = "executable")]
```
Comment (zanieb) [inline crates/uv-cli/src/lib.rs:2178]: Was there a particular motivation for adding an alias?

Comment (charliermarsh) [reply crates/uv-cli/src/lib.rs:2178]: In the moment it felt odd that we use the term "executable" everywhere but `--bin` for the directory. I'll remove it though.

File: crates/uv/src/commands/tool/dir.rs (hunk 2 above)
Comment (charliermarsh) [reply crates/uv/src/commands/tool/dir.rs:17]: Consistency with the rest of the file.

File: crates/uv/src/commands/tool/dir.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv/src/commands/tool/dir.rs:20]: Do you know why that tree of functions returns `Result` if it never returns `Err`?

File: crates/uv-cli/src/lib.rs (hunk 3 above)
Comment (charliermarsh) [reply crates/uv-cli/src/lib.rs:2178]: Fixed: https://github.com/astral-sh/uv/pull/5187

File: crates/uv/src/commands/tool/dir.rs (hunk 2 above)
Comment (charliermarsh) [reply crates/uv/src/commands/tool/dir.rs:17]: Fixed: https://github.com/astral-sh/uv/pull/5187

File: crates/uv/src/commands/tool/dir.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/src/commands/tool/dir.rs:20]: It was copied from the `Cache` implementation, but someone dropped the `Result` from there in https://github.com/astral-sh/uv/pull/3875

The tree should be updated. I can do it if you want.

**Pull request merged**